sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py

```diff
@@ -0,0 +1,350 @@
+from typing import Callable, Dict, Iterable, Optional, Tuple
+
+from feast import type_map
+from feast.data_source import DataSource
+from feast.errors import DataSourceNotFoundException
+from feast.feature_logging import LoggingDestination
+from feast.protos.feast.core.DataSource_pb2 import DataSource as DataSourceProto
+from feast.protos.feast.core.FeatureService_pb2 import (
+    LoggingConfig as LoggingConfigProto,
+)
+from feast.protos.feast.core.SavedDataset_pb2 import (
+    SavedDatasetStorage as SavedDatasetStorageProto,
+)
+from feast.repo_config import RepoConfig
+from feast.saved_dataset import SavedDatasetStorage
+from feast.value_type import ValueType
+
+
+class AthenaSource(DataSource):
+    def __init__(
+        self,
+        *,
+        timestamp_field: Optional[str] = "",
+        table: Optional[str] = None,
+        database: Optional[str] = None,
+        data_source: Optional[str] = None,
+        created_timestamp_column: Optional[str] = None,
+        field_mapping: Optional[Dict[str, str]] = None,
+        date_partition_column: Optional[str] = None,
+        query: Optional[str] = None,
+        name: Optional[str] = None,
+        description: Optional[str] = "",
+        tags: Optional[Dict[str, str]] = None,
+        owner: Optional[str] = "",
+    ):
+        """
+        Creates a AthenaSource object.
+
+        Args:
+            timestamp_field : event timestamp column.
+            table (optional): Athena table where the features are stored.
+            database: Athena Database Name
+            data_source (optional): Athena data source
+            created_timestamp_column (optional): Timestamp column indicating when the
+                row was created, used for deduplicating rows.
+            field_mapping (optional): A dictionary mapping of column names in this data
+                source to column names in a feature table or view.
+            date_partition_column : Timestamp column used for partitioning.
+            query (optional): The query to be executed to obtain the features.
+            name (optional): Name for the source. Defaults to the table_ref if not specified.
+            description (optional): A human-readable description.
+            tags (optional): A dictionary of key-value pairs to store arbitrary metadata.
+            owner (optional): The owner of the athena source, typically the email of the primary
+                maintainer.
+
+
+        """
+        # The default Athena schema is named "public".
+        _database = "default" if table and not database else database
+        self.athena_options = AthenaOptions(
+            table=table, query=query, database=_database, data_source=data_source
+        )
+
+        if table is None and query is None:
+            raise ValueError('No "table" argument provided.')
+        _name = name
+        if not _name:
+            if table:
+                _name = table
+            else:
+                warnings.warn(
```

Comment:
can you remove this? It's past Feast 0.21 now haha so should enforce there is a name

Reply:
I changed AthenaSource to have a name!